server/src/model.py

```diff
@@ -0,0 +1,62 @@
+import string
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics import accuracy_score, classification_report
+from nltk.corpus import stopwords
+from sklearn.ensemble import RandomForestClassifier
+
+
+# approach:
+# clean and preprocess the questions (remove punctuation, lowercase the text, etc.).
+# remove stop words like "is", "my", "the"
+# use TF-IDF (Term Frequency-Inverse Document Frequency) to convert the questions into numerical vectors
+# in comparison to CountVectorizer, CV just counts the frequency of a word in a given sentence/input
+# TF_IDF counts the frequency AND the Inverse Document Frequency
+# use Random Forest Classifier 
+# predict the intent of the user's question/input
+
+import re
+def model(question: string): 
```

Comment:
Can we split the training and the output so that we're not training the model everytime we need the output? This is inefficient. Consider saving the model into a joblib like we did in class.

Reply:
Right now, the function does preprocessing, training, and output all in one function. Let's clean up the comments and split the function into 3 (actually 2 because we already have a pre-processing function).